### ISSUE TITLE
Configure MetroAnimatedTabControl not to cache tabs

### ISSUE THREAD

**Issue by RayCarrot** [MahApps/MahApps.Metro#3652]
In my project I'm using the MetroAnimatedTabControl to contain all my tabs in my main window. Due to this the app is a bit slow to launch as every tab gets loaded on startup. I think it'd be great if you could configure how they get loaded. For example:
1. Load the tab only when the user first switches over to it. Then keep it in memory.
2. Load the tab when the user switches over it and then deload it when the user switches to another tab.

**Comment by punker76** [comment]
@RayCarrot Can you specify with which version of MahApps this happens? A short sample app would be also helpful.

**Comment by RayCarrot** [comment]
I'm not sure what you want an example of? This is just the current behavior of the animated tab control to always keep all tabs in memory.
The version I'm on is the latest one at NuGet that's not a pre-release, so 1.6.5.

**Comment by punker76** [comment]
@RayCarrot I got the same behavior with a normal TabControl inside a normal Window without MahApps.Metro.

**Comment by punker76** [comment]
@RayCarrot Here is my test sample

```xaml
<Window x:Class="WpfApp7.MainWindow"
        xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d"
        Title="MainWindow" Height="450" Width="800" WindowStartupLocation="CenterScreen">

    <Grid HorizontalAlignment="Center" VerticalAlignment="Center">
        <TabControl Style="{x:Null}">

            <TabItem Style="{x:Null}" Header="UserControl 1">
                <UserControl Loaded="UserContro1Loaded" />
            </TabItem>

            <TabItem Style="{x:Null}" Header="UserControl 2">
                <UserControl Loaded="UserContro2Loaded" />
            </TabItem>

            <TabItem Style="{x:Null}" Header="UserControl 3">
                <UserControl Loaded="UserContro3Loaded" />
            </TabItem>

        </TabControl>
    </Grid>

</Window>
```

```csharp
using System.Windows;

namespace WpfApp7
{
    /// <summary>
    /// Interaction logic for MainWindow.xaml
    /// </summary>
    public partial class MainWindow : Window
    {
        public MainWindow()
        {
            InitializeComponent();
        }

        private void UserContro1Loaded(object sender, RoutedEventArgs e)
        {
            MessageBox.Show("UserControl 1 Loaded");
        }

        private void UserContro2Loaded(object sender, RoutedEventArgs e)
        {
            MessageBox.Show("UserControl 2 Loaded");
        }

        private void UserContro3Loaded(object sender, RoutedEventArgs e)
        {
            MessageBox.Show("UserControl 3 Loaded");
        }
    }
}
```

![tabcontrol_loaded](https://user-images.githubusercontent.com/658431/67393556-ed860080-f5a2-11e9-8874-d3fceac47f5d.gif)

**Comment by RayCarrot** [comment]
I apologize, I must have remembered wrong how the WPF Tab Control works. I did some tests on my own and it does indeed load them all in at once. For some reason I remembered it only loading them in when you switched to the tab and thought the behavior had been changed with the MetroAnimatedTabControl.
One thing I realized is that the Tab Control you inherit from seems to have some special feature of keeping the tab content in the visual tree: https://github.com/ControlzEx/ControlzEx#TabControlEx
However it seems that this still loads them all at once, just prevents them from reusing the same panel.
Sorry for taking your time with this and thanks for the answer!
